clippy_lints/src/methods/mod.rs

```diff
@@ -1796,6 +1797,30 @@ declare_clippy_lint! {
     "replace `.splitn(2, pat)` with `.split_once(pat)`"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Check for `&mut iter` followed by a method call.
+    ///
+    /// ### Why is this bad?
+    /// This requires using parenthesis to signify precedence.
+    ///
+    /// ### Example
+    /// ```rust
+    /// let mut iter = ['a', 'b', '.', 'd'].iter();
+    /// let before_dot = (&mut iter).take_while(|&&c| c != '.').collect::<Vec<_>>();
+    /// let after_dot = iter.collect::<Vec<_>>();
```

Comment:
What does this `after_dot` example show?

Reply:
Nothing related to the lint. It's more of a practical use case for using `&mut iter` in the first place.

Reply:
Can you add a comment to the documentation then? Usually we only have good vs bad examples. Also that it is duplicated in the "use instead" example is a bit confusing IMO.

Reply:
Just removed the line.